source/Weapon.h

```diff
@@ -2,69 +2,105 @@
 #include <G3D/G3D.h>
 #include "ConfigFiles.h"
 
-class Projectile : public Entity {
-public:
-	shared_ptr<VisibleEntity>       entity;
+class Projectile : public VisibleEntity {
+protected:
+	// Timed mode
+	double							m_totalTime = 5.0;
+	// Propagation mode
+	bool							m_collision = false;
+	float							m_velocity = 0.0f;
+	float							m_gravity = 10.0f;
+	float							m_gravVel = 0.0f;
+	float							m_maxVel = 100.0f;
+	Point3							m_lastPos = Point3::zero();
 
-	Projectile() : m_totalTime(0) {}
-	Projectile(const shared_ptr<VisibleEntity>& e, RealTime t = 0) : entity(e), m_totalTime(t) {}
-	Projectile(const shared_ptr<VisibleEntity>& e, float velocity, bool collision = true, float gravity = 0.0f, RealTime t = 5.0) :
-		entity(e),
+	Projectile(const shared_ptr<VisibleEntity>& e, float velocity, bool collision = true, float gravity = 0.0f, SimTime t = 5.0) :
+		VisibleEntity(*e),
 		m_totalTime(t),
 		m_collision(collision),
 		m_velocity(velocity),
-		m_gravity(gravity){}
+		m_gravity(gravity) {}
+
+public:
+	Projectile() : m_totalTime(0) {}
```

Comment:
does this constructor need to be public, or could it be protected?

Reply:
I believe I got a compiler complaint when this wasn't public. 

I don't know the exact error off-hand, but I believe it had to do with construction. Though this might just be as simple as missing a void `static shared_ptr<Projectile> create()` for Array to initialize an `Array<shared_ptr<Projectile>>` from.